### ISSUE TITLE
authFilter get skipped on userAgent == "Nacos-Server"

### ISSUE THREAD

**Issue by flaneur2020** [alibaba/nacos#3237]
## Issue Description

Type: *bug report*

### Describe what happened (or what feature you want)

In the AuthFilter there's a logic about User-Agent check, if the User-Agent equals "Nacos-Server", the authFilter will be exited, so you can access every permission required API by simply setting User-Agent: "Nacos-Server". is this behaviour by design?

the code locates in: https://github.com/alibaba/nacos/blob/9a44b920a8b25cba3fdad5359921cd447a62ba6d/core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java#L69

```
        if (StringUtils.startsWith(userAgent, Constants.NACOS_SERVER_HEADER)) {
            chain.doFilter(request, response);
            return;
        }
```

### Describe what you expected to happen

the authentication can not be easily skipped by setting a plaintext http header. if there're requirements about accessing nacos servers by another nacos server, it'd be better to have a secret token.

**Comment by KomachiSion** [comment]
This is indeed a problem, we will enhance it.

**Comment by KomachiSion** [comment]
Fixed in 1.4.1
